code/controllers/subsystem/minimaps.dm

```diff
@@ -0,0 +1,281 @@
+	for(var/datum/minimap_updator/xenoupdators AS in update_targets[MINIMAP_STRING_XENO])
+		xenoupdators.minimap.overlays = minimaps_by_z["[xenoupdators.ztarget]"].xeno_images_raw.Copy()	//special handling for the first one in queue for that speed
+	for(var/datum/minimap_updator/marineupdators AS in update_targets[MINIMAP_STRING_MARINE])
+		marineupdators.minimap.overlays += minimaps_by_z["[marineupdators.ztarget]"].marine_images_raw
+	for(var/datum/minimap_updator/alphaupdators AS in update_targets[MINIMAP_STRING_ALPHA])
+		alphaupdators.minimap.overlays += minimaps_by_z["[alphaupdators.ztarget]"].alpha_images_raw
+	for(var/datum/minimap_updator/bravoupdators AS in update_targets[MINIMAP_STRING_BRAVO])
+		bravoupdators.minimap.overlays += minimaps_by_z["[bravoupdators.ztarget]"].bravo_images_raw
+	for(var/datum/minimap_updator/charlieupdators AS in update_targets[MINIMAP_STRING_CHARLIE])
+		charlieupdators.minimap.overlays += minimaps_by_z["[charlieupdators.ztarget]"].charlie_images_raw
+	for(var/datum/minimap_updator/deltaupdators AS in update_targets[MINIMAP_STRING_DELTA])
+		deltaupdators.minimap.overlays += minimaps_by_z["[deltaupdators.ztarget]"].delta_images_raw
+
+/**
+ * Adds an atom to the processing updators that will have blips drawn on them
+ * Arguments:
+ * * target: the target we want to be updating the overlays on
+ * * flags: flags for the types of blips we want to be updated
+ * * ztarget: zlevel we want to be updated with
+ */
+/datum/controller/subsystem/minimaps/proc/add_to_updaters(atom/target, flags, ztarget)
+	var/datum/minimap_updator/holder = new(target, ztarget)
+	if(flags & MINIMAP_FLAG_XENO)
+		LAZYADD(update_targets[MINIMAP_STRING_XENO], holder)
+	if(flags & MINIMAP_FLAG_MARINE)
+		LAZYADD(update_targets[MINIMAP_STRING_MARINE], holder)
+	if(flags & MINIMAP_FLAG_ALPHA)
+		LAZYADD(update_targets[MINIMAP_STRING_ALPHA], holder)
+	if(flags & MINIMAP_FLAG_BRAVO)
+		LAZYADD(update_targets[MINIMAP_STRING_BRAVO], holder)
+	if(flags & MINIMAP_FLAG_CHARLIE)
+		LAZYADD(update_targets[MINIMAP_STRING_CHARLIE], holder)
+	if(flags & MINIMAP_FLAG_DELTA)
+		LAZYADD(update_targets[MINIMAP_STRING_DELTA], holder)
+	updators_by_datum[target] = holder
+	update_targets_unsorted += target
+	RegisterSignal(target, COMSIG_PARENT_QDELETING, .proc/remove_updator)
+
+/**
+ * Removes a atom from the subsystems updating overlays
+ */
+/datum/controller/subsystem/minimaps/proc/remove_updator(atom/target)
+	SIGNAL_HANDLER
+	var/datum/minimap_updator/holder = updators_by_datum[target]
+	updators_by_datum -= target
+	LAZYREMOVE(update_targets[MINIMAP_STRING_XENO], holder)
+	LAZYREMOVE(update_targets[MINIMAP_STRING_MARINE], holder)
+	LAZYREMOVE(update_targets[MINIMAP_STRING_ALPHA], holder)
+	LAZYREMOVE(update_targets[MINIMAP_STRING_BRAVO], holder)
+	LAZYREMOVE(update_targets[MINIMAP_STRING_CHARLIE], holder)
+	LAZYREMOVE(update_targets[MINIMAP_STRING_DELTA], holder)
```

Comment:
cant you loop through the keys inside update_targets instead of removing manually ?

Reply:
Done